source/framefmt.c

```diff
@@ -202,6 +203,12 @@ static inline int serde_nonframed(
     struct aws_cryptosdk_framestate *AWS_RESTRICT state, struct aws_cryptosdk_frame *AWS_RESTRICT frame) {
     field_sized(state, &frame->iv, state->alg_props->iv_len);
     field_be64(state, &state->plaintext_size);
+
+    // Check that plaintext_size is within bounds so we avoid an overflow later
+    if (state->plaintext_size >= MAX_PLAINTEXT_SIZE) {
```

Comment:
This check prevents an arithmetic overflow in lines 212 or 213.

Reply:
AWS_ERROR_PRECONDITION once that is merged.

Reply:
Does it make sense to have a precondition in the middle of a function?